source/adapters/level_zero/kernel.cpp

```diff
@@ -41,6 +41,15 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueKernelLaunch(
         *OutEvent ///< [in,out][optional] return an event object that identifies
                   ///< this particular kernel execution instance.
 ) {
+  auto ZeDevice = Queue->Device->ZeDevice;
+
+  ze_kernel_handle_t ZeKernel{};
+  if (Kernel->ZeKernelMap.empty()) {
+    ZeKernel = Kernel->ZeKernel;
```

Comment:
Question: Could you confirm if there would ever be scenario where ZeKernelMap is empty, but ZeKernel is created? I see on Line 393 zeKernelCreate() is called and then added to this ZeKernelMap, is there other place ZeKernel would be created outside of being added to the map?

Reply:
the kernel map will never be empty, the map is the only place now where kernels can be created.

Reply:
Gotcha, so this branch exists for negative testing and should only be taken incase of a failure of memory allocation.